### ISSUE TITLE
When task description is too long an employee should be able to "read more"

### ISSUE THREAD

**Issue by j0nscalet** [j0nscalet/QualityKitchen.Mobile#21]
**Problem**
On the Main page when a task description is more than **3 lines** a "MORE" link should be displayed. When the user taps the more link the cell's height could expand so they can view the additional text. 

Here's a screenshot of an example task. As you can see it's too long and needs to be shortened:
![image](https://cloud.githubusercontent.com/assets/1328795/23927451/0378a96c-08d7-11e7-88e0-73e8cc4c561b.png)

A lot of apps do this, including instagram. Here's an example:
<img src="https://cloud.githubusercontent.com/assets/1328795/23927408/b7b9a6ca-08d6-11e7-8640-519c0d48bed2.PNG" width=150 height=225 />

**Solution**
- Xamarin Forms ListView is notorious for not updating ContentSize very well which gives me the feeling that we cannot achieve this without a rendering. (Maybe extend 'SwipableListView'?)

**Comment by j0nscalet** [comment]
@gaoguangli Here's your next issue. Before starting to code let's discuss the approach.

- Do you think we're going to have issues updating the ContentSize of the cell? (I've run into issues with the ContentSize of the cell not updating properly in the past. I use a RepeaterView to get around these issues)

Let me know your thoughts.
Thanks!

**Comment by gaoguangli** [comment]
I think contentsize will not working. we need to take another option. However, I will try.

**Comment by j0nscalet** [comment]
@gaoguangli that's what I thought. Let's not go the ListView route then because I think we'll get stuck with the content size. 

This is the reason why we have RepeaterView. It's based on a StackLayout and handle content updating much better than the ListView does. 

If you've got something quick you want to try with the ListView, feel free but I wouldn't spend too much time on it. 

The RepeaterView approach should work well. For an example, see [Incident Resolution Page](https://github.com/j0nscalet/QualityKitchen.Mobile/blob/dev/QualityKitchen.Mobile/Application/Pages/IncidentResolutionPage.xaml)

**Comment by j0nscalet** [comment]
@ershovd This issue is a bit more involved so I'd like to discuss your approach before starting! Thanks.

**Comment by ershovd** [comment]
@j0nscalet I've make some research and this issue (with dynamic row heigth) is more complex and seems not to be fully supported in X.Forms. We can try this one (viewcell renderer with row heigth calculation) https://forums.xamarin.com/discussion/comment/100947/#Comment_100947 
If it will not work, we can  try to implement Listview renderer similiar to https://forums.xamarin.com/discussion/comment/56693/#Comment_56693

**Comment by j0nscalet** [comment]
@ershovd sounds good. Yea I figured this issue will require some research. It's going to be a tricky one because ListView in XForms doesn't play very nice when you ask it to change cell height.

Something that may help you think about this clearly:
- How would this be done in Xamarin iOS native? (Button click that changes height of a uitableviewcell dynamically)
- Once that's clear then it may help you implement the renderer changes

Hopefully you're research turns up a good approach to this.

**Comment by j0nscalet** [comment]
Quick question about #21, looks like we're 3 hours into it, are you getting closer to a solution or farther away? 

I ask because we could simply wire up a tapped event there and display a dialog. Not exactly what we're looking for but it will work just fine. I've seen dialogs animate "up and down" from a cell to display from text (in the inbox by Google app) and then animate down. I don't think they're expanding the cell there but instead using a dialog. It would seem getting `uitableviewcell` to play nice is hard and we're not the only ones with that issue.

What do you think?

**Comment by ershovd** [comment]
@j0nscalet , ok, thanks for the feedback. I will try first the approach with settings height dynamically using ListViewrender (won't spend too much time on that). if this is not working, we'll follow your solution with dialog animation

**Comment by j0nscalet** [comment]
@ershovd cool, see how the dynamic heights go and let me know. I want to further detail the dialog approach before we head down that path.

**Comment by ershovd** [comment]
@j0nscalet  I am little bit stuck with this task. I spend 4+ hours today on it, tried the several approaches, but they are mostly now working. Currently I've switch to your approach (with separate view showing). I've impemented simple `DetailTaskPage` and viewmodel to display description on row selection. Looks ugly, but it least working. Furthermore, it seems that the only default animation (from bottom to top) suported in X.Forms (https://forums.xamarin.com/discussion/comment/69586/#Comment_69586 ). I've found this ready control, we can buy that (or create our own implementation of navigation page)
Now, initially I've focused on dynamic changing cell height. The call to `viewCell.ForceUpdateSize();` did changed the cell height, but in `ListViewDataSourceWrapper` the changing height not rendered properly (the row goes black  after trying to switch back to initial height). Furthermore, I've receive several calls to `GetCell` method with weird rowIndex, when clicking the cell.
So, I would like to hear feedback and maybe request some help, in which way to proceed. If is ok, from the UI point, I guess we should follow your proposed route (with dialog approach). I will clean up the code, make the layout looks nicely, and will add check for displaying only long-text description.
